watchdog-agent/src/main/java/io/github/algomaster99/Terminator.java

```diff
@@ -33,6 +34,9 @@ public byte[] transform(
 
     private static byte[] isLoadedClassWhitelisted(String className, byte[] classfileBuffer) {
         Map<String, List<Provenance>> fingerprints = options.getFingerprints();
+        if (RuntimeClass.isProxyClass(classfileBuffer) || RuntimeClass.isSynthetic(classfileBuffer)) {
```

Comment:
I am really unsure about synthetic classes. We should keep an eye on them and include them in the threat model.

Reply:
I would agree with you. We could remove this condition also, as I have never encountered a case yet. The test is also very artificial.